### PR TITLE
operationalize creating initial config file

### PR DESCRIPTION
**Pull request by gouthamr321**
This just takes chris's original script and operationalizes it and adds a few changes

notable changes:
- use an argument parser so things arent hard coded
- includes a helper function to sort round names
- Saves config file in json format instead of as a yaml format. This is because yaml files don't preserve the order of keys in a dictionary. See here: [https://github.com/yaml/pyyaml/issues/110](here). I'm not sure if saving this will cause issues downstream..... though should be easy to modify other scripts to read in the json file instead)

The output config files are here: /allen/aics/assay-dev/users/Goutham/4iProcessing-/4iprocessing/json_configs....... These look consistent to me compared to the original yaml files, but I'm hoping @cfrick13 can confirm

### CODE REVIEW

**Review by BrianWhitneyAI — APPROVED** [review]

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [inline 4iprocessing/generate_initial_config.py]: this line is superfluous i think

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [inline 4iprocessing/generate_initial_config.py]: lines 74 - 79 should be moved below line 84

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [inline 4iprocessing/generate_initial_config.py:101]: need to make this catch any round greater than 10. right now it only looks for round 11 but in theory we may have 12 or 13 rounds in the future.

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [inline 4iprocessing/generate_initial_config.py]: change name to `channels` instead of `zchannels`

Comment (cfrick13) [reply 4iprocessing/generate_initial_config.py]: I also want channels to be at the bottom, so maybe move this line below 107.

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [inline 4iprocessing/generate_initial_config.py]: typo: "defies"--> "defines"

**Review by cfrick13 — COMMENTED** [review]
made some comments in the code. looks good overall

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: Nice catch, line 24 is superfluous

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: This doesn't make any functional changes.... we are just defining the reference channel which doesn't affect the reading of the file..... but maybe I'm not understanding?

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: ohh, nice catch

File: 4iprocessing/generate_initial_config.py
Comment (cfrick13) [reply 4iprocessing/generate_initial_config.py]: yes. no functional changes. it just makes sense to put the code where we choose channels after we have determined the channels from reader.channels. so its just a cosmetic change.

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py:101]: Just confirming, here we only what this if statement to be executed if we are looking at the first round or the 20X timelapse.... is that correct?

Comment (cfrick13) [reply 4iprocessing/generate_initial_config.py:101]: correct

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: Addressed in new commits

**Comment by cfrick13** [comment]
we talked and decided to wait to merge this code until we finish cleaning up variable names in the code (in `FormattingAndSorting` branch). And also decided to look at other ways of writing yamls to violate alphabetical order

File: 4iprocessing/generate_initial_config.py
Comment (aditnath) [inline 4iprocessing/generate_initial_config.py]: might want to not have these hardcoded as these will change once the isilon is gone?

File: 4iprocessing/generate_initial_config.py
Comment (aditnath) [inline 4iprocessing/generate_initial_config.py]: we will need to include support for ome.tiffs as well, since those are the aligned files

**Review by aditnath — CHANGES_REQUESTED** [review]
left a couple comments, we might want to address the ome tiff functionality before merging

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: nice catch, took care of this

File: 4iprocessing/generate_initial_config.py
Comment (gouthamr321) [reply 4iprocessing/generate_initial_config.py]: also just took care of this one

**Comment by gouthamr321** [comment]
Incorporated some changes since last time. Notable changes are:
- workaround to save config file in yaml format without messing up order of elements in dictionary
- defined some operations as separate functions(e.g. determining which scenes to toss, determining reference channel)
- support for files saved in .tiff format for the future
- change to argument parser to not hardcode input directory files 

Not sure if this is ready to merge or if we still need to wait for FormattingAndSorting branch to be merged first? Thoughts?

**Comment by cfrick13** [comment]
We should be ready to merge this now. FormattingAndSorting branch is merged

**Review by BrianWhitneyAI — APPROVED** [review]

**Review by cfrick13 — APPROVED** [review]
looks good. Might just need change requests marked as completed

**Review by aditnath — APPROVED** [review]

**Comment by gouthamr321** [comment]
Operationalize config file generation

**Pull request merged**
